<launch>
  <arg name="camera_name" default="axis" />
  <arg name="hostname" default="192.168.0.90" />
  <arg name="username" default="root" />
  <arg name="password" default="" />
  <arg name="encrypt_password" default="true" />

  <arg name="enable_theora" default="0" />
  <arg name="enable_ptz" default="0" />
  <arg name="enable_defog" default="0" />
  <arg name="enable_ir" default="0" />
  <arg name="enable_wiper" default="0" />
  <arg name="width" default="640" />
  <arg name="height" default="480" />
  <arg name="camera" default="1" />

  <arg name="enable_ptz_teleop" default="0" />
  <arg name="joy_topic" default="/bluetooth_teleop/joy" />
  <arg name="joy_teleop_config" default="$(find axis_camera)/config/teleop_ps4.yaml" />

  <!-- desired FPS of the camera. 0=camera default -->
  <arg name="fps" default="0" />

  <group ns="$(arg camera_name)">
    <param name="hostname" value="$(arg hostname)" />
    <param name="username" value="$(arg username)" />
    <param name="password" value="$(arg password)" />
    <param name="use_encrypted_password" value="$(arg encrypt_password)" />
    <param name="width" value="$(arg width)" />
    <param name="height" value="$(arg height)" />
    <param name="camera" value="$(arg camera)" />
    <param name="frame_id" value="$(arg camera_name)_camera_link" />
    <param name="wiper" value="$(arg enable_wiper)" />
    <param name="defog" value="$(arg enable_defog)" />
    <param name="ir" value="$(arg enable_ir)" />
    <param name="fps" value="$(arg fps)" />
    <param name="ptz" value="$(arg enable_ptz)" />

    <node pkg="axis_camera" type="axis_camera_node" name="axis" />

    <node pkg="image_transport" type="republish" name="republish"
          args="compressed theora" if="$(arg enable_theora)">
      <remap from="in" to="image_raw" />
      <remap from="out" to="image_raw_out" />
    </node>

    <group if="$(eval arg('enable_ptz_teleop') and arg('enable_ptz'))">
      <node pkg="axis_camera" type="axis_teleop_node" name="axis_teleop">
        <rosparam command="load" file="$(arg joy_teleop_config)" />
        <remap from="joy" to="$(arg joy_topic)" />
      </node>
    </group>
  </group>
</launch>
